### PR TITLE
Move docs to OTP 27 ex_doc support

### LINKED ISSUE #133: OTP 27 docs
Yo! You folks interested in moving the docs to `-doc` docs from OTP 27? Simplifies the syntax quite a bit thanks to Markdown. I started working on this but figured I'd ask first 🙃

### PR DESCRIPTION
**Pull request by whatyouhide**
Went ahead with this! It looks gorgeous 😉 

Closes #133.

CleanShot 2024-07-10 at 18 04 14@2x

### CODE REVIEW

**Review by josevalim — DISMISSED** [review]

File: src/telemetry.erl [hunk 1]
```diff
@@ -78,26 +80,28 @@
...
-%%
-%% Note that you should not rely on the order in which handlers are invoked.
+?DOC("""
+Attaches the handler to many events.
+
+The handler will be invoked whenever any of the events in the `event_names` list is emitted. Note
+that failure of the handler on any of these invocations will detach it from all the events in
+`event_name` (the same applies to manual detaching using `detach/1`).
+
+<b>Note:</b> due to how anonymous functions are implemented in the Erlang VM, it is best to use
```
Comment (davydog187) [inline src/telemetry.erl:90]: Admonition block here would be a great change!

Comment (whatyouhide) [reply src/telemetry.erl:92]: Good catch, converted 🙃

**Review by josevalim — APPROVED** [review]

**Comment by josevalim** [comment]
:green_heart: :blue_heart: :purple_heart: :yellow_heart: :heart:

**Comment by wojtekmach** [comment]
Sorry to be the person but can we have a new release? :) I think this is a pretty huge deal, I believe for the very first time we have shell docs out of the box (without tools like docsh) for Erlang packages. This just works:

IMG_0333

**Comment by whatyouhide** [comment]
Lovely! I don't have maintainer privileges on Telemetry, I’m just a contributor so someone else will have to do 🥲

**Pull request merged**